platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_db_util.py

```diff
@@ -0,0 +1,602 @@
+            log_err(f'failed to delete reboot cause table entries due to {e}')
+
+    def _fetch_reboot_cause_list(self):
+        try:
+            REBOOT_CAUSE_TABLE_NAME = "REBOOT_CAUSE"
+            TABLE_NAME_SEPARATOR = '|'
+            db = SonicV2Connector(host='127.0.0.1')
+            db.connect(db.STATE_DB, False)
+            prefix = REBOOT_CAUSE_TABLE_NAME + TABLE_NAME_SEPARATOR
+            _hash = '{}{}'.format(prefix, '*')
+            table_keys = db.keys(db.STATE_DB, _hash)
+            reboot_causes = []
+            if table_keys is not None:
+                table_keys.sort(reverse=True)
+                table = []
+                for tk in table_keys:
+                    entry = db.get_all(db.STATE_DB, tk)
+                    name = tk.replace(prefix, "")
+                    cause = entry['cause'] if 'cause' in entry else ""
+                    time = entry['time'] if 'time' in entry else ""
+                    user = entry['user'] if 'user' in entry else ""
+                    comment = entry['comment'] if 'comment' in entry else ""
+                    reboot_causes.append(
+                        [name, cause, comment, time, user]
+                    )
+                return reboot_causes
+            else:
+                log_info("Reboot-cause history is not yet available in StateDB")
+                return []
+        except Exception as e:
+            log_err(f"Failed to fetch reboot cause list due to {e}")
+            return []
+
+    def update(self):
+        log_info("Start reboot cause list updating")
+        try:
+            stat = self._fetch_reboot_cause_list()
+            if stat == None:
+                raise Exception("Failed to fetch reboot cause list")
+            self._refresh_reboot_cause_db(stat)
+        except Exception as e:
+            log_err(f'failed to fetch reboot cause data due to {e}')
+        log_info("End reboot cause list updating")
+
+    def _refresh_reboot_cause_db(self, stat):
+        name = NOT_AVAILABLE
+        try:
+            if self.db == None:
+                return
+            self.delete_table_entries()
+            if len(stat) > 10:
+                stat = stat[:10]
+            for entry in stat:
+                table_name = f"{self.table_pattern}{entry[0]}"
+                self.db.hset(table_name, 'cause', entry[1])
+                self.db.hset(table_name, 'comment', entry[2])
+                self.db.hset(table_name, 'device', f"DPU{self.slot_id}")
+                self.db.hset(table_name, 'time', entry[3])
+                self.db.hset(table_name, 'user', entry[4])
+        except Exception as e:
+            log_err('Failed to update system health values for {} - {}'.format(name, repr(e)))
+
+class VersionUpdater():
+    def __init__(self, chassis, db):
+        self.db = db
+        self.chassis = chassis
+        self.slot_id = get_slot_id(self.chassis)
+        self.table = f'{VERSION_UPDATE_TABLE_NAME}|DPU{self.slot_id}|1'
```

Comment:
@shanshri what is this version table.? do we have HLD for this?

Reply:
This table passes all the dpu cpld/fw versions from DPU to NPU and is consumed by fwutil on the NPU side.   This platform specific and hence not captured in the HLD.

Reply:
@prgeor 
fwutil:
1. The plan is to keep the FW show/update/install for the  DPU managed components on the DPU side and NPU managed DPU related components on the NPU side and this will eliminate the need to exchange version information  from DPUs to NPU
2. The cleanup of "fwutil version_db_upate" can be completed in this PR
3. A separate  PR will be raise to cover the GoldFW update and install

Reply:
@prgeor, I have removed version_db_updater from here. Have added support for fwutil show, update and install for uboota, golduboot and emmc. For goldfw, will raise separate PR for update and install.